BlueprintUI/Sources/Layout/Stack.swift

```diff
@@ -560,50 +673,97 @@ extension StackLayout {
                 axis: .atMost(axisConstraint),
                 cross: crossConstraint)
             let constraint = vector.constraint(axis: axis)
-            let measuredSize = item.measure(in: constraint)
+            let measuredSize = item.content.measure(in: constraint)
 
             return measuredSize.cross(on: axis)
         }
 
-        // Then pick the max cross value based on the constraint
-        let maxCross: CGFloat
-        switch crossConstraint {
-        case .unconstrained:
-            maxCross = crossMagnitudes.reduce(0, max)
-        case .exactly(let exactConstraint):
-            maxCross = exactConstraint
-        case .atMost(let maxConstraint):
-            let maxMagnitude = crossMagnitudes.reduce(0, max)
-            maxCross = min(maxConstraint, maxMagnitude)
-        }
-
-        // Finally, form segments from the magnitudes and the alignment option
-        let segments = zip(items, crossMagnitudes).map { (item, measuredCross) -> Segment in
-            let origin: CGFloat
-            let magnitude: CGFloat
-
-            switch alignment {
-            case .center:
-                origin = (maxCross - measuredCross) / 2.0
-                magnitude = measuredCross
-
-            case .fill:
-                origin = 0.0
-                magnitude = maxCross
-
-            case .leading:
-                origin = 0.0
-                magnitude = measuredCross
-
-            case .trailing:
-                origin = maxCross - measuredCross
-                magnitude = measuredCross
+        func fillSegments() -> [Segment] {
+            // Determine the available cross based on the constraint
+            let availableCross: CGFloat
+            switch crossConstraint {
+            case .unconstrained:
+                availableCross = crossMagnitudes.reduce(0, max)
+            case .exactly(let exactConstraint):
+                availableCross = exactConstraint
+            case .atMost(let maxConstraint):
+                let maxMagnitude = crossMagnitudes.reduce(0, max)
+                availableCross = min(maxConstraint, maxMagnitude)
             }
 
-            return Segment(origin: origin, magnitude: magnitude)
+            // Form segments that fill the available space
+            let segments = Array(
+                repeating: Segment(origin: 0, magnitude: availableCross),
+                count: items.count
+            )
+
+            return segments
+        }
+
+        func alignSegments(to alignment: AlignmentID.Type) -> [Segment] {
+            // Get the alignment values for each child
+            let alignmentValues = items.indices.map { i -> CGFloat in
```

Comment:
Minor: Could also be a `zip`

Reply:
(Though I guess if there's three accesses in here; meh?)

Reply:
Yep, I had to give up the zip here since we're indexing 3 sequences.